lib/charms/prometheus_k8s/v0/prometheus_remote_write.py

```diff
@@ -892,14 +892,39 @@ def alerts(self) -> dict:
                 continue
             # Construct an ID based on what's in the alert rules
             error_messages = []
+            tool = CosTool(self._charm)
             for group in alert_rules["groups"]:
+
+                # Copy off rules so we don't modify an object we're iterating over
+                rules = group["rules"]
```

Comment:
This is unnecessary because we are not deleting elements.

Reply:
Yes, you're right.
This is the change that should be done in the library however (it's not my changes here - that's just new version of the lib). Fixing it here, won't help much, because it's just a copy of a lib.